crates/eth-rpc/src/lib.rs

```diff
@@ -25,7 +25,7 @@ pub enum RpcError {
 ///
 /// Will return `Err` if an error occurs when running the `ServerBuilder` start fails.
 pub async fn run_server<P: Provider + Send + Sync + 'static>(
```

Comment:
I believe we should keep the same signature for run_server, and trigger the build beforehand *** C.F below comment

Reply:
Done.